examples/jsm/loaders/PLYLoader.js

```diff
@@ -408,8 +408,40 @@ class PLYLoader extends Loader {
 			}
 
 			if ( buffer.colors.length > 0 ) {
+				if ( buffer.colors.length == buffer.vertices.length  ){
+					geometry.setAttribute( 'color', new Float32BufferAttribute( buffer.colors, 3 ) );
+				}else{
+					const numVertices =  buffer.vertices.length;
+					const vertexColors = new Float32BufferAttribute(new Float32Array(numVertices * 3), 3);
+					vertexColors.needsUpdate = true;
+
+
+
+					// aversing the triangle surface, 
+					// copying each vertex color to the corresponding index position, 
+					// and copying each vertex color to the corresponding index position
+					for (let i = 0, il = geometry.index.count; i < il; i += 3) {
+						const a = geometry.index.array[i];
+						const b = geometry.index.array[i + 1];
+						const c = geometry.index.array[i + 2];
+
+
+						const faceColor = new Color();
+						faceColor.fromArray(buffer.colors, i);
+
+
+
+						vertexColors.setXYZ(a, faceColor.r, faceColor.g, faceColor.b);
+						vertexColors.setXYZ(b, faceColor.r, faceColor.g, faceColor.b);
+						vertexColors.setXYZ(c, faceColor.r, faceColor.g, faceColor.b);
+
+
+					}
+					geometry.setAttribute('color', vertexColors);
+					geometry.setAttribute('FaceColor',  new Float32BufferAttribute( buffer.colors, 3 ) );
```

Comment:
The usual workflow for supporting face color definitions is to convert an indexed-geometry to a non-indexed geometry and then create a new vertex color attribute. So we do the same like for `faceVertexUvs`. I suggest you update the PR as follows:

- Save face color data in a new data structure similar to face uvs. Name it `faceVertexColors`. 
- Do not just save the single color but already expand it for three vertices. Meanin:
```js
buffer.faceVertexColors.push( _color.r, _color.g, _color.b );
buffer.faceVertexColors.push( _color.r, _color.g, _color.b );
buffer.faceVertexColors.push( _color.r, _color.g, _color.b );
```
You can then update the following section to honor the new color data:
```js
if ( buffer.faceVertexUvs.length > 0 ) {

    geometry = geometry.toNonIndexed();
    geometry.setAttribute( 'uv', new Float32BufferAttribute( buffer.faceVertexUvs, 2 ) );

}
```

Reply:
Thank you for your guidance and help, let me benefit a lot, because I am a beginner, can not even JavaScript, write a little bad.